src/lib/optimizer/join_ordering/join_graph_converter.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <map>
+#include <memory>
+#include <optional>
+#include <set>
+#include <vector>
+
+#include "join_graph.hpp"
+#include "logical_query_plan/union_node.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+
+/**
+ * Turns an LQP into a JoinGraph.
+ *
+ * Consider this simple LQP:
+ * [0] [Projection] x1, y1
+ *  \_[1] [Predicate] x2 <= y1
+ *     \_[2] [Cross Join]
+ *        \_[3] [MockTable] x
+ *        \_[4] [MockTable] y
+ *
+ * The JoinGraph created from it would contain two vertices (x and y), one edge (between x and y, with the predicate
+ * x2 <= y1) and one parent_relation (Projection, left child side)
+ *
+ * The name "Converter" stems from the fact that nodes representing predicates and joins are physically removed from the
+ * LQP. To turn a JoinGraph back into a LQP, use a JoinOrdering algorithm such as DpCcp.
```

Comment:
Why do you have to remove things from the LQP? In my imagination, the removal part should happen in the optimizer rule once we decided on a join order, not at the time when we build some auxiliary structure our join order algorithm happens to need. Also, this class feels like some weird type of constructor that is none. We should probably discuss this in person

Reply:
You also have `static JoinGraph JoinGraph::from_predicates` in addition to the regular constructors. To me, the construction responsibilities are unclear.